source/common/common/hash.h

```diff
@@ -58,7 +61,7 @@ class MurmurHash {
 private:
   static inline uint64_t unalignedLoad(const char* p) {
     uint64_t result;
-    memcpy(&result, p, sizeof(result));
+    safe_memcpy<uint64_t, const uint64_t>(&result, reinterpret_cast<const uint64_t*>(p));
```

Comment:
I don't think you need to give the template args explicitly. The compiler knows what types the args are. So this can be just:
```
safe_memcpy(&result, reinterpret_cast<const uint64_t*>(p));
```
Here and elsewhere in the PR.